command.go

```diff
@@ -1556,12 +1677,12 @@ func (cmd *XInfoConsumersCmd) readReply(rd *proto.Reader) error {
 func readXConsumerInfo(rd *proto.Reader) (XInfoConsumer, error) {
 	var consumer XInfoConsumer
 
-	n, err := rd.ReadArrayLen()
+	n, err := rd.ReadMapLen()
```

Comment:
This is a good example of what I mean about `ReadMapReply`. We can even add a helper like `ReadFixedMapLen(3)` that will check that map has the expected number of entries.